### PR TITLE
Updated Readthedocs documentation in line with current practices.

### PR DESCRIPTION
**Pull request by Kat-90**
Addresses https://github.com/metoppv/mo-blue-team/issues/795

This PR updates the best practices sections on the Readthedocs site in line with our current ways of working.
To build the webpages locally attempt the following:
- Activate the conda environment 
- Navigate to the improver top directory and enter: bin/improver-tests doc
- Then: firefox doc/build/html/index.html

### CODE REVIEW

**Review by brhooper — COMMENTED** [review]
Thanks @Kat-90. I've made some suggestions, mostly for rephrasing sentences to aid readability. Feel free to ignore them if you don't think they represent improvements.

File: doc/source/Guidance-for-Reviewing-Code.rst [hunk 1]
```diff
@@ -80,6 +75,7 @@ As a Reviewer:
 
         * Does the code do what is supposed to?
         * Are errors handled appropriately?
+        * Is the code written to be run efficiently?
```
[the remark this answers is not in the document]
Comment (Kat-90) [reply doc/source/Guidance-for-Reviewing-Code.rst]: Thanks, I've changed both.

**Review by brhooper — APPROVED** [review]
Thanks for these changes @Kat-90. I'm happy to approve this now.

**Review by Katie-Howard — DISMISSED** [review]
This all looks fine now, so I'm happy to approve the changes 👍

**Review by Katie-Howard — APPROVED** [review]
It looks like the tests all pass now so I'm happy with these changes 👍

**Pull request merged**